src/views/activity/activity.view.tsx

```diff
@@ -33,6 +33,7 @@ const Activity: FC = () => {
   const [lastLoadedItem, setLastLoadedItem] = useState(0);
   const [total, setTotal] = useState(0);
   const [wrongNetworkBridges, setWrongNetworkBridges] = useState<Bridge["id"][]>([]);
+  const [finaliseDisabledBridges, setFinaliseDisabledBridges] = useState<Bridge["id"][]>([]);
```

Comment:
Have you tried to keep the state locally on each `BridgeCard` component instead of having a map of the disabled ones in the parent component?

Reply:
mmm... the state would be kept since the component would not be re-rendered because the `key={bridge.id}` would not change. Let me try!

Reply:
mmm... wait no, we can't do that. The actual problem is that the user may cancel the TX from Metamask and in such a case, we have to reset this state and we don't have access to the `catch` from the `BridgeCard`. Does this make sense to you?

(Sorry for the spam I should probably think a bit before I start answering comments...)

Reply:
yep, makes sense! didn't see that part of the code